### PR TITLE
feat(component): color-picker

### PR DESCRIPTION
**Pull request by ErKeLost**

### CODE REVIEW

File: packages/varlet-ui/src/color-picker/components/color-picker-canvas/color-picker-canvas.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          },
+        }
+        Dragger.triggerDragEvent(paletteInstance.vnode.el as HTMLElement, dragConfig)
+      }
+    })
+
+    function renderCanvas() {
+      if (canvasElement.value) {
+        const canvas = canvasElement.value.getContext('2d')
+        if (canvas) {
```
Comment (songjianet) [inline packages/varlet-ui/src/color-picker/components/color-picker-canvas/color-picker-canvas.tsx:67]: if (!canvasElement.value) return
const canvas = canvasElement.value.getContext('2d')
if (canvas) return

File: packages/varlet-ui/src/color-picker/components/color-picker-hue-slider/color-picker-hue-slider.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    function handleClickSlider(event: Event) {
+      if (event.target !== barElement.value) {
+        handleMoveBar(event as MouseEvent)
+      }
+    }
+
+    return () => {
+      return (
+        <div class={[n(), props.disabled ? n('disabled') : null]}>
```
Comment (songjianet) [inline packages/varlet-ui/src/color-picker/components/color-picker-hue-slider/color-picker-hue-slider.tsx:96]: props.disabled && n('disabled')

File: packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              <div class={n('swatch')}>
+                {swatch.map((colorItem) => {
+                  const hsv = HexToHSV(colorItem)
+                  const iconColor = changeColorValue(hsv!, 0.5)
+                  return (
+                    <div class={n('color')} onClick={() => call(props['onUpdate:color'], hsv)}>
+                      <div style={{ background: colorItem }}>
+                        {color!.value && deepEqual(color!.value, hsv) ? (
+                          <var-icon size="15" name="checkbox-marked-circle" color={iconColor.color} />
+                        ) : undefined}
```
Comment (songjianet) [inline packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx:36]: {color!.value && deepEqual(color!.value, hsv) &&

**Review by songjianet — CHANGES_REQUESTED** [review]

File: packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx (hunk 3 above)
Comment (ErKeLost) [reply packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx:36]: copy that

Comment (ErKeLost) [reply packages/varlet-ui/src/color-picker/components/color-picker-swatches/color-picker-swatches.tsx:36]: del this branch restart new pr

**Pull request closed without merging**